src/drivers/actuators/voxl_esc/voxl_esc.cpp

```diff
@@ -94,23 +94,140 @@ VoxlEsc::~VoxlEsc()
+
+		while( (!got_response) && (hrt_elapsed_time(&t_request) < t_timeout) ){
+			px4_usleep(100); //sleep a bit while waiting for ESC to respond
+
+			int nread = _uart_port->uart_read(_read_buf, sizeof(_read_buf));
+
+			for (int i = 0; i < nread; i++) {
+				int16_t parse_ret = qc_esc_packet_process_char(_read_buf[i], &_fb_packet);
+
+				if (parse_ret > 0) {
+					hrt_abstime response_time = hrt_elapsed_time(&t_request);
+					//PX4_INFO("got packet of length %i",ret);
+					_rx_packet_count++;
+					uint8_t packet_type = qc_esc_packet_get_type(&_fb_packet);
+					uint8_t packet_size = qc_esc_packet_get_size(&_fb_packet);
+
+					if (packet_type == ESC_PACKET_TYPE_VERSION_EXT_RESPONSE && packet_size == sizeof(QC_ESC_EXTENDED_VERSION_INFO)) {
+						QC_ESC_EXTENDED_VERSION_INFO ver;
+						memcpy(&ver, _fb_packet.buffer, packet_size);
+						
+						PX4_INFO("VOXL_ESC: \tESC ID     : %i", ver.id);
+						PX4_INFO("VOXL_ESC: \tBoard Type : %i: %s", ver.hw_version, board_id_to_name(ver.hw_version).c_str());
+
+						uint8_t *u = &ver.unique_id[0];
+						PX4_INFO("VOXL_ESC: \tUnique ID  : 0x%02X%02X%02X%02X%02X%02X%02X%02X%02X%02X%02X%02X",
+							 u[11], u[10], u[9], u[8], u[7], u[6], u[5], u[4], u[3], u[2], u[1], u[0]);
+
+						PX4_INFO("VOXL_ESC: \tFirmware   : version %4d, hash %.12s", ver.sw_version, ver.firmware_git_version);
+						PX4_INFO("VOXL_ESC: \tBootloader : version %4d, hash %.12s", ver.bootloader_version, ver.bootloader_git_version);
+						PX4_INFO("VOXL_ESC: \tReply time : %uus",(uint32_t)response_time);
+						PX4_INFO("VOXL_ESC:");
+
+						if (ver.id == esc_id){
+							memcpy(&_version_info[esc_id],&ver,sizeof(ver));
+							got_response = true;
+						}
+					}
+				}
+			}
+		}
+
+		if (!got_response){
+			PX4_ERR("VOXL_ESC: ESC %d version info response timeout", esc_id);
+		}
+	}
+
+	//check the SW version of the ESCs
+	bool esc_detection_fault = false;
+	for (int esc_id=0; esc_id < VOXL_ESC_OUTPUT_CHANNELS; esc_id++){
+		if (_version_info[esc_id].sw_version == 0){
+			PX4_ERR("VOXL_ESC: ESC ID %d was not detected", esc_id);
+			esc_detection_fault = true;
+		}
+	}
+
+	//check the firmware hashes to make sure they are the same. Firmware hash has 8 chars
+	for (int esc_id=1; esc_id < VOXL_ESC_OUTPUT_CHANNELS; esc_id++){
+		if (strncmp(_version_info[0].firmware_git_version,_version_info[esc_id].firmware_git_version, 8) != 0) {
+			PX4_ERR("VOXL_ESC: ESC %d Firmware hash does not match ESC 0 firmware hash:  %.12s != %.12s", 
+				esc_id, _version_info[esc_id].firmware_git_version, _version_info[0].firmware_git_version);
+			esc_detection_fault = true;
+		}
 	}
 
-	//get_instance()->ScheduleOnInterval(10000); //100hz
+	//if firmware version is equal or greater than VOXL_ESC_EXT_RPM, ESC packet with extended rpm range is supported. use it
+	_extended_rpm      = true;
+	for (int esc_id=0; esc_id < VOXL_ESC_OUTPUT_CHANNELS; esc_id++){
+		if (_version_info[esc_id].sw_version < VOXL_ESC_EXT_RPM)
+			_extended_rpm = false;
```

Comment:
Please put parentheses around all "if" statements, even if they only have one line.

Reply:
done! (plus another minor tweak). also tested on the bench